src/pages/home/ReportScreen.js

```diff
@@ -1,54 +1,109 @@
-import ONYXKEYS from '../../ONYXKEYS';
+import {fetchActions} from '../../libs/actions/Report';
 
 const propTypes = {
-    // id of the most recently viewed report
-    currentlyViewedReportID: PropTypes.string,
-};
+    /* Navigation api provided by react navigation */
+    navigation: PropTypes.shape({
+        /* Display the drawer programmatically */
+        openDrawer: PropTypes.func.isRequired,
+    }).isRequired,
 
-const defaultProps = {
-    currentlyViewedReportID: '0',
+    /* Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
+        /* Route specific parameters used on this screen */
+        params: PropTypes.shape({
+            /* The ID of the report this screen should display */
+            reportID: PropTypes.string,
+        }).isRequired,
+    }).isRequired,
 };
 
-const ReportScreen = (props) => {
-    const activeReportID = parseInt(props.currentlyViewedReportID, 10);
-    if (!activeReportID) {
-        return null;
+class ReportScreen extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+            error: null,
+        };
     }
 
-    return (
-        <ScreenWrapper
-            style={[
-                styles.appContent,
-                styles.flex1,
-                styles.flexColumn,
-            ]}
-        >
-            <HeaderView
-                reportID={activeReportID}
-                onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
-            />
-            <View style={[styles.dFlex, styles.flex1]}>
-                <ReportView reportID={activeReportID} />
-            </View>
-        </ScreenWrapper>
-    );
-};
+    componentDidMount() {
+        this.fetchReport();
+    }
+
+    componentDidUpdate(prevProps) {
+        // Reports changed, reset and load new data
+        if (this.props.route.params.reportID !== prevProps.route.params.reportID) {
+            this.fetchReport();
+        }
+    }
+
+    get canRenderMainContent() {
+        return Boolean(this.reportID) && !this.state.isLoading && !this.state.error;
+    }
+
+    get reportID() {
+        const params = this.props.route.params;
+        return parseInt(params.reportID, 10) || 0;
+    }
+
+    fetchReport() {
+        if (!this.reportID) { return; }
+
+        // This adds a small fade in transition when the loader appears
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+
+        this.setState({
+            isLoading: true,
+            error: null,
+        });
+
+        fetchActions(this.reportID)
+            .catch((error) => {
+                /* Todo: do something here if the user is going to be left with an empty view
+                *   E.g. Alert like "Oops, we couldn't load this conversation. Please Try again"
+                *     button Retry -> this.fetchReport() */
+                this.setState({error});
```

Comment:
This is pretty much against the initial design that we are going for. That said, we haven't fully thought out how error handling will work. The conversation will happen eventually and maybe our original intentions will be challenged, but for now - if the report doesn't load then it doesn't load.

Reply:
Yep, it would be a rare case to end up in this catch, and even then there might still be some chat history available